src/routes/search.rs

```diff
@@ -1,29 +1,33 @@
-use axum::{extract::State, Extension, Form};
+use axum::{
+    extract::{Path, State},
+    http::StatusCode,
+    Form,
+};
 use maud::{html, Markup};
 use serde::Deserialize;
 use std::sync::Arc;
 use tracing::{debug, instrument};
 
-use crate::{components, middlewares::CookieUserState};
+use crate::{components, scraper::Term};
 
-use super::DatabaseAppState;
+use super::{AppError, DatabaseAppState};
 
 #[derive(Deserialize, Debug)]
 pub struct Search {
-    #[allow(dead_code)]
-    search: String, // FIXME: after we actually implement add_to_calendar
+    search: String,
 }
 
-#[instrument(level = "debug")]
+#[instrument(level = "debug", skip(state))]
 pub async fn search(
-    State(_): State<Arc<DatabaseAppState>>,
-    Extension(_): CookieUserState,
+    Path(id): Path<String>, // TODO: implement Deserialize on Term
+    State(state): State<Arc<DatabaseAppState>>,
     Form(query): Form<Search>,
-) -> Markup {
-    debug!("search");
-    let result: Vec<String> = Vec::new();
+) -> Result<Markup, AppError> {
+    let term = id.parse::<Term>().map_err(|_| StatusCode::BAD_REQUEST)?;
+    let courses = state.search(term, &query.search)?;
```

Comment:
what happens if this fails? i was expecting to see a `.map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;`

Reply:
`AppError` is already doing that, the `anyhow::Error` variant just logs the error and returns 500:
https://github.com/brennanmcmicking/scheduler/blob/ad98a155ca2efa75e1049111b21423f47f24e8be/src/routes.rs#L101-L113

Reply:
I suppose there is a longer-term conversation to be had about how we should handle fatal errors. An error in the database code or serialization logic for example can't really be handled, so we could also consider just panicking and using [`tower_http::catch_panic`](https://docs.rs/tower-http/latest/tower_http/catch_panic/index.html) middleware. But they say not to rely on it and to prefer `Result` instead.

It looks like [CloudFlare is just doing what we do](https://github.com/cloudflare/freighter/blob/main/crates/freighter-api-types/src/index/error.rs), implementing their own `Error` enums that implement `IntoResponse`, and then have a field for `From<anyhow::Error>` for catch-all errors, so I say we just keep doing this.